azure-pipelines.yml

```diff
@@ -0,0 +1,80 @@
+stages:
+ - stage: check
+   displayName: Compilation check
+   dependsOn: []
+   jobs:
+     - template: azure/cargo-check.yml@templates
+       parameters:
+         name: cargo_check
+ # This represents the minimum Rust version supported.
+ # Tests are not run as tests may require newer versions of rust.
+ - stage: msrv
+   displayName: "Minimum supported Rust version"
+   dependsOn: []
+   jobs:
+     - template: azure/cargo-check.yml@templates
+       parameters:
+        rust: 1.34.0
+ - stage: custom
+   displayName: "Special tests"
+   dependsOn: check
+   pool:
+     vmImage: ubuntu-16.04
+   jobs:
+     - job: custom_stable
+       displayName: "Stable custom tests"
+       steps:
+         - template: azure/install-rust.yml@templates
+           parameters:
+            rust: stable
+         - bash: cd tracing/test-log-support && cargo test
+           displayName: "Test log support"
+         - bash: cd tracing/test_static_max_level_features && cargo test
+           displayName: "Test static max level features"
+     - job: custom_nightly
+       displayName: "Nightly custom tests"
+       continueOnError: true
+       steps:
+         - template: azure/install-rust.yml@templates
+           parameters:
+            rust: nightly
+         - script: cargo test --features=doctest-readme --all
+           displayName: "Test README as doctest"
+         - bash: cd tracing-futures/test_std_future && cargo test
+           displayName: "Test std::future support for tracing-futures"
+ - stage: style
+   displayName: Style linting
+   dependsOn: ${{ format('{0}check', parameters.prefix) }}
+   jobs:
+    - template: azure/rustfmt.yml@templates
+      parameters:
+        name: rustfmt
+    - template: azure/rustfmt.yml@templates
+      parameters:
+        name: rustfmt_beta
+        rust: beta
+        allow_fail: true
+    - template: azure/cargo-check.yml@templates
+      parameters:
+        name: warning_free
+        err_on_warn: true
+        allow_fail: true
+ - stage: test
+   displayName: Test suite
+   dependsOn: check
+   jobs:
+     - template: azure/tests.yml@templates
+ - stage: coverage
+   displayName: Code coverage
+   dependsOn: test
+   jobs:
+     - template: azure/coverage.yml@templates
+       parameters:
+         codecov_token: $(CODECOV_TOKEN_SECRET)
+
+resources:
+  repositories:
+    - repository: templates
+      type: github
+      name: crate-ci/azure-pipelines
```

Comment:
Just for my understanding, does this mean that our CI build is now dependent on the existence of this repo?

Reply:
That is correct. You can point at a particular commit using `ref:` if you're worried about long-term stability. See also https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#using-other-repositories and https://github.com/crate-ci/azure-pipelines/issues/15#issuecomment-511031151.

Reply:
Okay, cool, thanks. I'm fine with this (assuming the `crate-ci` repos are going to uphold a reasonable stability guarantee), but I just wanted to be aware of all the cases where something _could_ break the build.